LanguageFeatures/Control-flow-collections/Null-aware-elements/runtime_A01_t01.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If element is a `nullAwareExpressionElement` with expression `e`:
+/// - Evaluate `e` to `v`.
+///   - If `v` is not `null` then append it to result. Else the `null` is
+///     discarded.
+///
+/// @description Check a runtime semantics of a `nullAwareExpressionElement` in
+/// set collections
+/// @author sgrekhov22@gmail.com
+
+import '../../../Utils/expect.dart';
+
+main() {
+  var e1 = 2 > 1 ? 1 : null;
+  var e2 = 1 > 2 ? 2 : null;
+
+  var set1 = {
+    ?e1,
+    ?null,
+    ?e2
+  };
+  Expect.setEquals({e1}, set1);
+
+  var set2 = {
+    ?e1 + 1,
```

Comment:
Isn't this just going to be a compile-time error? (`e1` has type `int?`).

Perhaps there is an extension operator `+` that works on an `int?`? Otherwise we could declare it, in order to enable `e1 + 1` (and, in general, `?e` where `e` is various kinds of expressions).